src/widget.rs

```diff
@@ -89,8 +89,21 @@ impl<'a> Renderer<'a> {
         let lnum_len = num_digits(lines_len);
         let bottom_row = cmp::min(top_row + height, lines_len);
         let mut lines = Vec::with_capacity(bottom_row - top_row);
+        // big enough for the largest line I can imagine
+        const STARS: &str = "******************************\
+                             ******************************\
+                             ******************************\
+                             ******************************\
+                             ******************************\
+                             ******************************";
+        const STARSLEN: usize = STARS.len();
         for (i, line) in self.0.lines()[top_row..bottom_row].iter().enumerate() {
-            lines.push(self.0.line_spans(line.as_str(), top_row + i, lnum_len));
+            let mut thisline = line.as_str();
+            if self.0.password_mode {
+                let count = thisline.chars().count();
+                thisline = &STARS[0..cmp::min(count, STARSLEN)];
+            }
+            lines.push(self.0.line_spans(thisline, top_row + i, lnum_len));
```

Comment:
consider perhaps:
```rust
let line = if self.0.password_mode {
    "*".repeat(line.len())
} else {
    line
};
```
Note: this probably should do unicode `width()` rather than `len()`, but I don't see that in the library already.

Reply:
Oh I tried that. But the returned Text object is marshalling `&str`s. So the strings must persist after the function returns. FYI - see here. https://stackoverflow.com/questions/77269247/trying-to-subsitiute-text-of-variable-size?noredirect=1#comment136223144_77269247